rotkehlchen/chain/evm/tokens.py

```diff
@@ -96,9 +96,16 @@ def generate_multicall_chunks(
 
 
 class EvmTokens(metaclass=ABCMeta):
-    def __init__(self, database: 'DBHandler', evm_inquirer: 'EvmNodeInquirer'):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_inquirer: 'EvmNodeInquirer',
+            # mapping of owner addresses to proxy addresses
+            proxies_mapping: Optional[dict[ChecksumEvmAddress, ChecksumEvmAddress]] = None,
```

Comment:
You don't need this argument though. Why pass this?

The way you are currently doing this, it won't work.

The only place you initialize `EthereumTokens` is the constructor of `EthereumManager` and there you do:

```python
            tokens=EthereumTokens(
                database=node_inquirer.database,
                ethereum_inquirer=node_inquirer,
                proxies_mapping=node_inquirer.proxies_inquirer.get_accounts_having_proxy(),
            ),
```

This means that they will never ever be requeried, and if they do they won't be reflected in the tokens. So you throw the entire requery/cache logic out of the window.

Why do this? In fact you are passing the `evm_inquirer` as the argument above. So you don't need to do this **at all**.